client/src/util/stateManager/matrix.ts

```diff
@@ -184,7 +184,7 @@ function encodeDictArray(
   const [dCodes, dDict] = _getCodesAndDictVectorOffset(builder, uType, uData);
   builder.startObject(2);
   builder.addFieldOffset(0, dCodes, 0);
-  builder.addFieldOffset(0, dDict, 0);
+  builder.addFieldOffset(1, dDict, 0);
   return builder.endObject();
```

Comment:
Small but important: the dict-encoded column encoder now puts the dictionary in field 1 (was mistakenly 0), which fixes the serialized format the backend expects.